### PR TITLE
CB-17731 Filter upgrade candidate images by image name availability.

### PR DESCRIPTION
**Pull request by pkedvessy**
See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageNameUpgradeImageFilter.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        return ORDER_NUMBER;
+    }
+
+    private List<Image> filterImages(ImageFilterResult imageFilterResult, ImageCatalogPlatform imageCatalogPlatform, String cloudPlatform, String region) {
+        return imageFilterResult.getImages()
+                .stream()
+                .filter(image -> {
+                    try {
+                        imageService.determineImageName(cloudPlatform, imageCatalogPlatform, region, image);
+                        return  true;
```
Comment (Bajzathd) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageNameUpgradeImageFilter.java:55]: typo: double spaces

File: core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageNameUpgradeImageFilter.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                    } catch (CloudbreakImageNotFoundException e) {
+                        LOGGER.error(e.getMessage());
```
Comment (Bajzathd) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageNameUpgradeImageFilter.java:57]: This seems like an expected use case, so I think it should be info or warn, and please add an explanation message

File: core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageNameUpgradeImageFilter.java (hunk 1 above)
Comment (pkedvessy) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageNameUpgradeImageFilter.java:55]: its not a typo, it is missing checkstlye rule :-D

File: core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageNameUpgradeImageFilter.java (hunk 2 above)
Comment (pkedvessy) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageNameUpgradeImageFilter.java:57]: it should be an error in produciotn but an acceptable edge case in our test environment. so I think the error is valid.

**Review by Bajzathd — APPROVED** [review]

**Review by bergerdenes — APPROVED** [review]
